src/Analysis/pipeline/parameter_types.py

```diff
@@ -0,0 +1,323 @@
+import abc
+
+class BaseData(metaclass=abc.ABCMeta):
+    """
+    Abbstract class od port types, that define user for
+    validation and translation to string
+    """
+    def __init__(self):
+        self.name = ""
+        """Display name of port"""
+        self.description = ""
+        """Display description of port"""
+    
+    @abc.abstractmethod 
+    def to_string(self, value):
+        """Presentation of type in json yaml"""
+        pass
+
+    def contain(self, data):
+        """
+        return if structure contain set data
+        """
+        return True
+        
+    @abc.abstractmethod 
+    def is_set(self):
+        """
+        return if structure contain real data
+        """
+        pass
+    
+    @abc.abstractmethod 
+    def assigne(self):
+        """
+        Assigne appropriate python variable to data     
+        """
+        pass
+        
+    @abc.abstractmethod 
+    def getter(self):
```

Comment:
Taky jsem o tom tak přemýšlel, ale zvolil jsem název podle typu protože:
- při složitějších strukrurách, by to byla úroven navíc a úplně nevím jak se mi podaří ošetřit přiřazení (názvy proměnné by mohli napovědět)
- s přiřazením je v pythonu problém:
        struct.a = 5 - mohu ošetřit a zkontrolovat (dělám to)
        struct.a = Int(5) - mohu ošetřit a zkontrolovat (dělám to)
        a=5 - asi neošetřím (v pythonu se přiřadí reference na objekt a nic jako operátor přiřazení neexistuje ) musí se psát a=Int(5)
       takže pokud to někdo strčí přímo do dat pomocí struct.a.data='pět', nezkontroluju to a zdá se mě lepší at to stčí do struct.a.integer='pět', když na něj pak nedopatřením vybafne python chyba, třeba to bude srozumitelnější  
 - tím že budou proměnné jinak pojmenovány se mohu vyhnout některým chybám (záměnám) při psaní kódu
- jediný kód navíc je funkce getter, která tam navíc u složitějších proměnných bude muset být kvůli transformaci, takže toho kódu navíc moc není

Pro toto řešení jsem se spíš přiklonil jako k méně špatnému, asi celou strategii zadávání budeme muset probrat slovně, takže to tak zatím nechávám.